cinema/serializers.py

```diff
@@ -68,13 +93,58 @@
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionSerializer(read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+    def validate(self, data):
+        data = super().validate(data)
+        ticket = Ticket(
+            movie_session=data.get("movie_session"),
+            row=data.get("row"),
+            seat=data.get("seat"),
+        )
+        ticket.full_clean()
+        return data
+
+
+class TicketRowSeatSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketRowSeatSerializer(
+        many=True,
+        read_only=True,
+        source="tickets"
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=True, allow_null=False)
+
+    class Meta:
+        model = Order
+        fields = ("id", "created_at", "tickets")
+
+    def create(self, validated_data):
+        tickets_data = validated_data.pop("tickets")
```

Comment:
The `create` method in `OrderSerializer` attempts to pop 'tickets' from `validated_data`, but 'tickets' is marked as `read_only=True` in the serializer, meaning it won't be present in `validated_data`. This will raise a `KeyError`. Consider revising this logic to handle the creation of tickets appropriately.